rules/windows/process_creation/win_susp_rundll32_activity.yml

```diff
@@ -11,27 +11,33 @@ tags:
     - attack.execution  # an old one 
     - attack.t1218.011
     - attack.t1085      # an old one 
-author: juju4
+author: juju4, Jonhnathan Ribeiro, oscd.community
 date: 2019/01/16
 logsource:
     category: process_creation
     product: windows
 detection:
     selection:
-        CommandLine:
-            - '*\rundll32.exe* url.dll,*OpenURL *'
-            - '*\rundll32.exe* url.dll,*OpenURLA *'
-            - '*\rundll32.exe* url.dll,*FileProtocolHandler *'
-            - '*\rundll32.exe* zipfldr.dll,*RouteTheCall *'
-            - '*\rundll32.exe* Shell32.dll,*Control_RunDLL *'
-            - '*\rundll32.exe javascript:*'
-            - '* url.dll,*OpenURL *'
-            - '* url.dll,*OpenURLA *'
-            - '* url.dll,*FileProtocolHandler *'
-            - '* zipfldr.dll,*RouteTheCall *'
-            - '* Shell32.dll,*Control_RunDLL *'
-            - '* javascript:*'
-            - '*.RegisterXLL*'
+        CommandLine|contains:
+            - 'url.dll*OpenURL'
+            - 'url.dll*OpenURLA'
+            - 'url.dll*FileProtocolHandler'
+            - 'zipfldr.dll*RouteTheCall'
+            - 'shell32.dll*Control_RunDLL'
+            - 'shell32.dll*ShellExec_RunDLL'
+            - 'javascript:'
+            - '.RegisterXLL'
+            - 'mshtml.dll*PrintHTML'
+            - 'advpack.dll*LaunchINFSection'
+            - 'advpack.dll*RegisterOCX'
+            - 'ieadvpack.dll*LaunchINFSection'
+            - 'ieadvpack.dll*RegisterOCX'
+            - 'ieframe.dll*OpenURL'
+            - 'shdocvw.dll*OpenURL'
+            - 'syssetup.dll*SetupInfObjectInstallAction'
+            - 'setupapi.dll*InstallHinfSection'
+            - 'pcwutl.dll*LaunchApplication'
+            - 'dfshim.dll*ShOpenVerbApplication'
```

Comment:
Hello @w0rk3r! 

Could you please change this section to the following:

```
        - CommandLine|contains:
            - 'javascript:'
            - '.RegisterXLL'
        - CommandLine|contains|all:
            - 'url.dll'
            - 'OpenURL'
        - CommandLine|contains|all:
            - 'url.dll'
            - 'OpenURLA'
        - CommandLine|contains|all:
            - 'url.dll'
            - 'FileProtocolHandler'
        - CommandLine|contains|all:
            - 'zipfldr.dll'
            - 'RouteTheCall'
        - CommandLine|contains|all:
            - 'shell32.dll'
            - 'Control_RunDLL'
        - CommandLine|contains|all:
            - 'shell32.dll'
            - 'ShellExec_RunDLL'
        - CommandLine|contains|all:
            - 'mshtml.dll'
            - 'PrintHTML'
        - CommandLine|contains|all:
            - 'advpack.dll'
            - 'LaunchINFSection'
        - CommandLine|contains|all:
            - 'advpack.dll'
            - 'RegisterOCX'
        - CommandLine|contains|all:
            - 'ieadvpack.dll'
            - 'LaunchINFSection'
        - CommandLine|contains|all:
            - 'ieadvpack.dll'
            - 'RegisterOCX'
        - CommandLine|contains|all:
            - 'ieframe.dll'
            - 'OpenURL'
        - CommandLine|contains|all:
            - 'shdocvw.dll'
            - 'OpenURL'
        - CommandLine|contains|all:
            - 'syssetup.dll'
            - SetupInfObjectInstallAction'
        - CommandLine|contains|all:
            - 'setupapi.dll'
            - 'InstallHinfSection'
        - CommandLine|contains|all:
            - 'pcwutl.dll'
            - 'LaunchApplication'
        - CommandLine|contains|all:
            - 'dfshim.dll'
            - 'ShOpenVerbApplication'
```

The thing is that some LM/SIEM systems don't process '*' as Splunk or Elasticsearch, so it won't work out of the box for them, and it would be hard for a SIEM expert to understand how exactly he/she supposes to construct a correlation rule looking at that detection logic.

Thank you!

Reply:
Just updated the rule, nice!